bd_spot_wrapper/spot_wrapper/spot.py

```diff
@@ -340,6 +350,79 @@ def get_image_responses(self, sources, quality=None, pixel_format=None):
+        DO NOT SUPPORT DEPTH and RGB_IN_DEPTH.
+        """
+        source_list = []  # type: List[str]
+
+        if not camera_sources:
+            print("Empty list passed in logger camera sources")
+
+        # Verification checks
+        for camera_source in camera_sources:
+            if (
+                "_depth" in camera_source
+                and "_depth_in_visual_frame" not in camera_source
+                and "_depth_in_hand_color_frame" not in camera_source
+            ):
+                print(
+                    f"Invalid source name {camera_source}. Camera source should either be rgb or depth aligned in rgb"
+                )
+            else:
+                source_list.append(camera_source)
+
+        self.source_list = source_list  # TODO define in init
+        print(f"Initialized logging for sources : {self.source_list}")
+
+    def update_logging_data(self):
+
+        log_packet = {
+            "timestamp": time.time(),
+            "camera_data": [],
+            "vision_T_base": None,
+            "base_pose_xyt": None,
+            "arm_pose": None,
+        }  # type: Dict[str, Any]
+
+        for camera_source in self.source_list:
+            img_responses = self.get_image_responses([camera_source])
+            log_packet["camera_data"].append(
+                {
+                    "src_info": camera_source,
+                    "raw_image": image_response_to_cv2(
+                        img_responses[0]
+                    ),  # np.ndarray ... How to save? (Serialize to bytes / List / PNG)?
+                    "camera_intrinsics": self.get_camera_intrinsics_as_3x3(
+                        img_responses[0].source.pinhole.intrinsics
+                    ),  # np.ndarray
+                    "base_T_camera": self.get_sp_SE3_spot_a_T_b(
+                        img_responses[0].shot.transforms_snapshot,
+                        frame_a="body",
+                        frame_b=spot_cam_frame_names[camera_source],
+                    ).matrix(),  # np.ndarray  .. pickle to save
+                }
+            )
+
+        log_packet["vision_T_base"] = self.get_sp_SE3_spot_a_T_b(
+            frame_tree_snapshot=None, frame_a="vision", frame_b="body"
+        ).matrix()  # np.ndarray  .. pickle to save
+        log_packet["base_pose_xyt"] = np.asarray(
+            self.get_xy_yaw()
+        )  # redundant? Check if it is consistent with vision_T_base
+        log_packet["arm_pose"] = self.get_arm_joint_positions()
+        # TODO: Add Gripper states (Griper Open/Close) : How to get gripper state from BD?
+        log_packet["is_gripper_closed"] = bool(
+            self.robot_state_client.get_robot_state().manipulator_state.is_gripper_holding_item
+        )
+        # TODO: Add force in gripper. How to get force reading?
+
+        # TODO: Check for optimized solution to store mixed DS in memory
+        # TODO: .to_list -> json || pickle dump to .json / .gz.json
```

Comment:
I think just pickle is fine
Don't need this to be json, those are usually larger in size and images, etc don't need to be in json?

Reply:
`import pickle`

`with open('output.pkl', 'wb') as pickle_file:
     pickle.dump(log_packet, pickle_file)
    `

Reply:
Thanks, added!